canonical-web-and-design/maas-ui
# fix(ui): minor improvements to subnet Used IP table

This branch turns the IP node text into links and also handles
extra cases for IP usage.

## Done

- Updated subnet "Used IP addresses" table to include:
  - Rendering the node as link, if it exists
  - Extra cases for IP usage 
  - Make "IP address" column text wrap if it gets long

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the details page of a subnet with used IP addresses that are being used for a node
- Check that the "Node" column is rendered as a link
- Open the console and edit the "IP address" text to be super long and check that it wraps to the next line instead of truncating

## Fixes

Fixes #1286

## Screenshot

![2022-02-11_14-35](https://user-images.githubusercontent.com/25733845/153538131-e55ab64b-df04-4e0a-95c2-407e033dcb86.png)
